### PR TITLE
Add subcommand: edit fix

### PR DESCRIPTION
**Pull request by Liujingfang1**

### CODE REVIEW

File: pkg/commands/edit/fix/fix_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import (
+	"strings"
+	"testing"
+
+	"sigs.k8s.io/kustomize/pkg/fs"
+)
+
+func TestFix(t *testing.T) {
+	fakeFS := fs.MakeFakeFS()
+	fakeFS.WriteTestKustomization()
```
Comment (monopole) [inline pkg/commands/edit/fix/fix_test.go]: Uh - our test kustomization should have the required fields.

Should fix that.

So this test could make a string missing those fields, and call WriteTestKustomizationWith(blah) instead.

File: pkg/commands/edit/fix/fix.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/pkg/commands/kustfile"
+	"sigs.k8s.io/kustomize/pkg/fs"
+)
+
+// NewCmdFix returns an instance of 'fix' subcommand.
+func NewCmdFix(fSys fs.FileSystem) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "fix",
```
Comment (monopole) [inline pkg/commands/edit/fix/fix.go]: Should also modify the log messages at
https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/types/kustomization.go#L157
to tell user to run this command.

Comment (Liujingfang1) [reply pkg/commands/edit/fix/fix.go]: updated the log messages

File: pkg/commands/edit/fix/fix_test.go (hunk 1 above)
Comment (Liujingfang1) [reply pkg/commands/edit/fix/fix_test.go]: > Uh - our test kustomization should have the required fields.

fixed that.

Also updated the function used here.

**Pull request merged**